refocus-bdk-server.js

```diff
@@ -188,46 +204,74 @@ module.exports = (config) => {
 
     const socket = io.connect(connectUrl, opts);
 
-    socket.on(initalizeEventName, (data) => {
+    socket.on(initalizeEventName, async (data) => {
       const eventData = JSON.parse(data);
       const room = eventData[initalizeEventName];
+      const { id, updatedAt } = room;
+      const hasAlreadyBeenConsumed = cache &&
+        await cache.hasBeenConsumed(id, updatedAt, botName, initalizeEventName);
+      if (hasAlreadyBeenConsumed) return;
       app.emit('refocus.internal.realtime.bot.namespace.initialize', room);
-      log.realtime('New Room', room);
     });
 
-    socket.on(settingsChangedEventName, (data) => {
+    socket.on(settingsChangedEventName, async (data) => {
       const eventData = JSON.parse(data);
       const room = eventData[settingsChangedEventName];
+      const { id, updatedAt } = room;
+      const hasAlreadyBeenConsumed = cache &&
+        await cache.hasBeenConsumed(id, updatedAt, botName,
+          settingsChangedEventName);
+      if (hasAlreadyBeenConsumed) return;
       app.emit('refocus.room.settings', room);
     });
 
-    socket.on(botActionsAdd, (data) => {
+    socket.on(botActionsAdd, async (data) => {
       const eventData = JSON.parse(data);
       const action = eventData[botActionsAdd];
+      const { id, updatedAt } = action;
+      const hasAlreadyBeenConsumed = cache &&
+        await cache.hasBeenConsumed(id, updatedAt, botName, botActionsAdd);
+      if (hasAlreadyBeenConsumed) return;
       app.emit('refocus.bot.actions', action);
     });
 
-    socket.on(botActionsUpdate, (data) => {
+    socket.on(botActionsUpdate, async (data) => {
```

Comment:
I'd put the same code of those 4 .on in one method that can be called as if (hasAlreadyBeenConsumed(param1, ...)) - they are the same, just the 3rd argument changes